langserver/hover.go

```diff
@@ -44,7 +44,15 @@ func (h *LangHandler) handleHover(ctx context.Context, conn JSONRPC2Conn, req *j
 		// TODO(sqs): make this be like (T).F not "struct field F string".
 		s = "struct " + o.String()
 	} else if o != nil {
-		s = types.ObjectString(o, qf)
+		if obj, ok := o.(*types.TypeName); ok {
+			typ := obj.Type().Underlying()
+			if _, ok := typ.(*types.Struct); ok {
+				s = "type " + obj.Name() + " struct"
+			}
```

Comment:
I tried it just now, and the answer is no. The actual problem is that `obj.Type().Underlying().String()` is actually what returns the 'concrete type' i.e. `struct{...all fields...}` instead of just `struct` which reads a lot better.

```
--- FAIL: TestServer (0.04s)
    --- FAIL: TestServer/go_detailed (0.00s)
        --- FAIL: TestServer/go_detailed/hover-a.go:1:17 (0.00s)
        	langserver_test.go:560: got "type T struct{F string}", want "type T struct"
```